python-package/lightgbm/dask.py

```diff
@@ -1141,7 +1141,7 @@ def __init__(
         )
 
     _base_doc = LGBMClassifier.__init__.__doc__
-    _before_kwargs, _kwargs, _after_kwargs = _base_doc.partition('**kwargs')
+    _before_kwargs, _kwargs, _after_kwargs = _base_doc.partition('**kwargs')  # type: ignore
```

Comment:
the warnings mentioned in the description come from the fact that `LGBMClassifier.__init__.__doc__` has type `Optional[str]`.

Proposing that these be ignored, for the reasons described in https://github.com/microsoft/LightGBM/pull/4837#discussion_r758927420